### ISSUE TITLE
Trying to ecode a float into a uint64 field results in a badarith exception instead of gpb_type_error

### ISSUE THREAD

**Issue by kjellwinblad** [tomas-abrahamsson/gpb#228]
First of all, thanks for making this useful library. While using it within the [EMQX MQTT message broker](https://github.com/emqx/emqx), we noticed the following issue:

When using `ProtobufModule:encode_msg` to encode a map into a protobuf binary, and the value of one filed has an incorrect type one normally gets a nice `gpb_type_error` exception but when the expected type is uint64 and the actual type is a float one instead gets a `badarith` exception.

**How to reproduce**

(Probably it is easy to reproduce this in a more simple way if one is more familiar with gpb and protobuf than I am):

1. Download the sparkplug_b schema https://raw.githubusercontent.com/emqx/emqx/541d03a0ba776861e081b1c69728d05f3e5bec4d/apps/emqx_schema_registry/priv/sparkplug_b.proto
2. Run the following code in the shell after starting with `rebar3 shell` in the root of the project (replace the path to `sparkplug_b.proto` with the correct one first):
```
{ok, Data} = file:read_file("/my/path/to/sparkplug_b.proto"),
Source = erlang:iolist_to_binary(Data),
{ok, my_mod, BeamBinary} = gpb_compile:string(
    my_mod,
    Source,
    [
        binary,
        strings_as_binaries,
        {maps, true},
        {maps_oneof, flat},
        {verify, always},
        {maps_unset_optional, omitted}
    ]
),
{module, my_mod} = code:load_binary(my_mod, "memory", BeamBinary).
```
3. Now run the following in the same shell:
```
my_mod:encode_msg(#{seq => 139395, uuid => 2}, 'Payload').
```
4. The above gives the following exception which is expected since `uuid` should be a string.
```
** exception error: {gpb_type_error,{bad_unicode_string,[{value,2},
                                                         {path,"Payload.uuid"}]}}
     in function  my_mod:mk_type_error/3 (my_mod.erl, line 3513)
     in call from my_mod:v_msg_Payload/3 (my_mod.erl, line 3435)
     in call from my_mod:encode_msg/3 (my_mod.erl, line 190)
```
5. Now run the following:
```
my_mod:encode_msg(#{seq => 139395.5, uuid => <<"hello">>}, 'Payload').
```
6. In this case we get the following badarith exception but I would have expected to get a similar gpb_type_error as above. Notice that the `seq` field has the type uint64 but we assigned it the float value 139395.5.
```
** exception error: an error occurred when evaluating an arithmetic expression
     in operator  band/2
        called as 139395.5 band 127
     in call from my_mod:e_varint/2 (my_mod.erl, line 739)
     in call from my_mod:encode_msg_Payload/3 (my_mod.erl, line 503)
```

**Comment by tomas-abrahamsson** [comment]
Oh, indeed. For integer fields, the generated code just checked the range `Min =< N, N =< Max`, but failed to check that the value was also an integer, so floats in the range accidentally slipped through. This is now corrected. Thanks for making me aware.

The fix is included in just-released 4.19.9, so I'll be closing this ticket.

**Comment by kjellwinblad** [comment]
Thanks @tomas-abrahamsson, for the very quick response!
